introspection-engine/connectors/sql-introspection-connector/src/introspection/enums.rs

```diff
@@ -83,7 +75,18 @@ fn render_enum<'a>(r#enum: EnumPair<'a>, output: &mut OutputContext<'a>) -> rend
         }
 
         if variant.name().is_empty() || sanitize_datamodel_names::needs_sanitation(&variant.name()) {
+            let warning = warnings::EnumAndValue {
+                enm: r#enum.name().to_string(),
+                value: variant.name().to_string(),
+            };
+
+            output.warnings.enum_values_with_empty_names.push(warning);
             rendered_variant.comment_out();
+        } else if variant.mapped_name().is_some() {
+            remapped_values.push(warnings::EnumAndValue {
+                value: variant.name().to_string(),
+                enm: r#enum.name().to_string(),
+            });
         }
```

Comment:
Recapping to make sure I understand:  in the first branch, we call `needs_sanitation` twice (once in `.name()`, once in this this conditional) to figure out what `name()` did, because it doesn't return that anymore. In the second branch, we push a warning about remapped values and we know if the remapping happened because of reintrospection or because of sanitization because we only have the else if the first branch didn't match. I think it's worth a comment if understand this right.

Reply:
That is correct.